### DISCUSSION TITLE
[SOLVED] Advanced search operations

### DISCUSSION THREAD

**Discussion by CsatiZoltan** [martinrotter/rssguard#397] [Ideas]
Hi!

I have suggestions for searching feeds (new feature in da352303356f1c5a2f76d4a287e2aadf5751fd09). Currently, when you enter a phrase, the search is done in a case-sensitive manner. What about making the feed search case-insensitive, just as how the message search works?

My second idea is to enable searching based on regex (the `QRegularExpression` class is available as of Qt5) or by other method, such as glob pattern matching.

**Comment by martinrotter** [comment]
@CsatiZoltan Done in 190e6d12.

Also note that at this point filtering of messages only takes TITLE of messages into account. Would it make sense to also make filtering work with contents of messages?

**Reply by CsatiZoltan** [reply]
Wow, that was fast!

I tried the regex, and it works fine.

> Would it make sense to also make filtering work with contents of messages?

I think that that would be extremely useful. In release dc988bf4 it worked (see the screenshot below), but in the new release 190e6d1, it does not.
![ksnip_20210413-204713](https://user-images.githubusercontent.com/12742890/114605456-677bdf00-9c9a-11eb-8e38-fac456c0e186.png)

**Reply by martinrotter** [reply]
Yes, there are some Qt library quirks when it comes to fine grained filtering. There is some default behavior which sort of works except it sometimes does not. When you want to make filtering somehow specific, it is sometimes pain in the ass to get things working.

**Reply by martinrotter** [reply]
OK, here it is: c59daa04.

It now performs filtering across ALL message view columns. Test and let me know.

**Reply by CsatiZoltan** [reply]
I confirm that simple and regex search work both in the feeds and in the messages. Moreover, both simple and regex search work for searches *within* (i.e. in the contents of) the messages.
For me, these additions greatly improved the speed I can locate the message I want.

**Comment by martinrotter** [comment]
Allright I am marking this as solved. :)
